MekHQ/src/mekhq/gui/utilities/ExportXML.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * ExportXML.java
+ * 
+ * MegaMek - Copyright (C) 2018 - The MegaMek Team
+ * 
+ * This file is part of MekHQ.
+ * 
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.gui.utilities;
+
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ResourceBundle;
+
+import javax.swing.JTable;
+
+import megamek.common.logging.LogLevel;
+import mekhq.MekHQ;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.CampaignOptions;
+import mekhq.campaign.parts.Part;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.SkillType;
+import mekhq.campaign.personnel.SpecialAbility;
+import mekhq.gui.CampaignGUI;
+import mekhq.gui.GuiTabType;
+import mekhq.gui.PersonnelTab;
+import mekhq.gui.WarehouseTab;
+import mekhq.gui.model.PartsTableModel;
+
+
+public class ExportXML implements IExport
+{
+    @Override
+    public String exportTable(JTable table, File file) {
+        //TODO
+        return null;
+    }
+
+    @Override
+    public String exportFinances(File file, String format, Campaign c) {
+        //TODO
+        return null;
+    }
+
+    @Override
+    public String exportParts(File file, CampaignGUI gui) {
+        final String METHOD_NAME = "exportParts()";
+        String report = "";
+        if (gui.getTab(GuiTabType.WAREHOUSE) != null) {
+            try {
+                JTable partsTable = ((WarehouseTab)gui.getTab(GuiTabType.WAREHOUSE)).getPartsTable();
+                PartsTableModel partsModel = ((WarehouseTab)gui.getTab(GuiTabType.WAREHOUSE)).getPartsModel();
+                int row = partsTable.getSelectedRow();
+                if (row < 0) {
+                    MekHQ.getLogger().log(IExport.class, METHOD_NAME, LogLevel.WARNING,
+                            "ERROR: Cannot export parts if none are selected! Ignoring."); //$NON-NLS-1$
+                    return resourceMap.getString("dlgNoPartsSelected.text");
+                }
+                Part selectedPart = partsModel.getPartAt(partsTable
+                        .convertRowIndexToModel(row));
+                int[] rows = partsTable.getSelectedRows();
```

Comment:
I'm almost sure `JTable.getSelectedRows()` will return  a length-one array if only one part is selected: you could do without `selectedPart` (and the related logic below)